web/modules/custom/lit_cover_service/src/Service/BatchService.php

```diff
@@ -0,0 +1,252 @@
+      $context['sandbox']['progress']++;
+      $context['sandbox']['current_node'] = $node->nid;
+    }
+
+    self::deleteFileEntities($deletedImageFiles);
+
+    // Inform the batch engine that we are not finished,
+    // and provide an estimation of the completion level we reached.
+    if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
+      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
+    }
+  }
+
+  /**
+   * Batch Finished callback.
+   *
+   * @param bool $success
+   *   Success of the operation.
+   * @param array $results
+   *   Array of results for post processing.
+   * @param array $operations
+   *   Array of operations.
+   */
+  public function batchFinished($success, array $results, array $operations) {
+    $messenger = \Drupal::messenger();
+    if ($success) {
+      $messenger->addMessage(t('@count books processed.', ['@count' => count($results)]));
+    }
+    else {
+      // An error occurred.
+      // $operations contains the operations that remained unprocessed.
+      $error_operation = reset($operations);
+      $messenger->addMessage(
+        t('An error occurred while processing @operation with arguments : @args',
+          [
+            '@operation' => $error_operation[0],
+            '@args' => print_r($error_operation[0], TRUE),
+          ]
+        )
+      );
+    }
+  }
+
+  /**
+   * Clear the cover image field
+   *
+   * @param \Drupal\node\Entity\Node $node
+   *
+   * @return \Drupal\file\Entity\File|null
+   */
+  private function clearImageField(Node $node): ?File {
+    /** @var \Drupal\file\Plugin\Field\FieldType\FileFieldItemList $deletedImageField */
+    $deletedImageField = $node->get('field_book_cover_image');
+    $file = $deletedImageField->entity;
+    $node->set('field_book_cover_image', NULL);
+
+    return $file;
+  }
+
+  /**
+   * Set the image file
+   *
+   * @param \Drupal\node\Entity\Node $node
+   */
+  private function setImageFile(Node $node): void {
+    $isbn = $node->get('field_book_isbn')->getString();
+    if ($isbn) {
+      $coverService = \Drupal::service('lit_cover_service.cover_service');
+      $coverImage = $coverService->getCoverImage($isbn);
+
+      if ($coverImage) {
+        $imageField = [
+          'target_id' => $coverImage->id(),
+          'alt' => '',
+          'title' => $node->getTitle(),
+        ];
+        $node->set('field_book_cover_image', $imageField);
+      }
+    }
+  }
+
+  /**
+   * Delete files
+   *
+   * @param \Drupal\file\Entity\File[] $files
+   */
+  public function deleteFileEntities(array $files): void {
+    $files = array_filter($files);
+
+    try {
+      /** @var EntityTypeManagerInterface $entityManager */
+      $entityManager = \Drupal::service('entity.manager');
+      $storage = $entityManager->getStorage('file');
+      $storage->delete($files);
+    } catch (InvalidPluginDefinitionException $e) {
+      // @TODO handle
```

Comment:
@TODO's?

Reply:
Not sure how to best handle exceptions here. Stop and crash or just log it for debugging.

Reply:
Just log it to console... so the one that executes the command knowns not all is well

Reply:
Updated the PR to do `\Drupal::logger('lit_cover_service')->error($e->getMessage());`

Reply:
Trouble is the code will run on `node_save` and as ajax callback, not just as console commands